### PR TITLE
Perl doc

### PR DESCRIPTION
**Pull request by brodieG**
This does the same thing as the [Perl pull request](https://github.com/MangoTheCat/rematch2/pull/11), but it also updates documentation to be more consistent with the addition of the argument.  Because the functions are so closely related and the edits that I was making applied to most of them, I merged documentation / cross referenced sections.  I also fixed a few typos along the way.  Since the changes I made are substantial and may not be in line with your style, I submit these as a separate PR in case you don't want to incorporate them.

### CODE REVIEW

**Comment by gaborcsardi** [comment]
Hmmm, I saw the other one first, and I merged it.....

**Comment by gaborcsardi** [comment]
And this one conflicts, of course....

**Comment by brodieG** [comment]
I'll fix this and re-submit in the evening.

**Comment by brodieG** [comment]
@gaborcsardi this should now be mergeable.

**Comment by brodieG** [comment]
Wow this is like the 10th time I've hit the "close and comment" instead of comment button.  Maybe just IMO, but terrible UI.  Or maybe I'm just stupid.

File: R/exec-all.R [hunk 1]
```diff
@@ -1,56 +1,9 @@
+#' @include exec.R
```
Comment (gaborcsardi) [inline R/exec-all.R]: Do we really need this?

**Comment by gaborcsardi** [comment]
Thanks! I like most of this, but maybe I would keep the `re_exec` and `re_exec_all` manual pages separate. I think the functions are similar, but also different, so to avoid confusion, it is just best to document each on its own page.

Do you mind updating the PR? Alternatively, I can also merge it now, and change it later....

**Comment by brodieG** [comment]
Sure, I'll update this evening.

**Comment by brodieG** [comment]
@gaborcsardi this should do it.  I tried to deduplicate as much of the docs as possible, but for some reason `@inherit` does not inherit description.  I can't be bothered debugging further so I just copied the block.

File: DESCRIPTION [hunk 2]
```diff
@@ -16,3 +16,10 @@ Suggests:
     covr,
     testthat
 Encoding: UTF-8
+Collate:
```
Comment (gaborcsardi) [inline DESCRIPTION:19]: This is not needed I think.

**Pull request merged**